rust/src/io/commit.rs

```diff
@@ -422,7 +422,7 @@ pub(crate) async fn commit_transaction(
             write_config,
         )?;
 
-        debug_assert_eq!(manifest.version, target_version);
+        manifest.version = target_version;
```

Comment:
This assert was failing because `manifest.version` here comes from `dataset.manifest` which is the old version.  I think it's safe to just latch onto the version we got from the commit (`dataset.manifest.version` should be the read version and not the destination version).

Reply:
Seems reasonable.